Reduce Bottom Spacing for Seamless Panels

EssWhyy opened the pull request:
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [x] Others, please explain:

Fixes #2222

<!--
  If this pull request is addressing an issue, link to the issue: #xxx

  If this pull request completely addresses an issue, use one of the closing keywords: "Fixes #xxx" or "Resolves #xxx"
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**
I have modified the CSS modifiers for the panel cards button and bottom hr tags to reduce the empty space at the bottom of the panels. Intended effect is shown below. 

![possible panel css fix](https://github.com/MarkBind/markbind/assets/39799639/3cbc23ba-bc2b-423a-a2f6-6f83d557a972)


**Anything you'd like to highlight/discuss:**
I am not certain if the hr tag css I changed in NestedPanel.vue is the correct css. Since bootstrap.min.css also has a css for hr tags, but I'm not allowed to change it or it will fail tests.   

**Testing instructions:**
NA

<!--
  Any special testing instructions **not including** our automated tests.
-->

**Proposed commit message: (wrap lines at 72 characters)**

<!--
  See this link for more info on how to write a good commit message:
  https://se-education.org/guides/conventions/git.html

  As best as possible, write a succinct commit title in 50 characters

|---------This is the width of 50 chars----------|
|-----------This is the width of 72 chars for your reference-----------|
-->

Reduce margin spacing for hr tag within seamless panels

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes <!-- It's tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

tlylt commented:
Thanks, @EssWhyy for tackling this.

> ![image](https://github.com/MarkBind/markbind/assets/41845017/d7040dbc-1036-417f-93bf-c90b7d54511c)

1. Image is broken

> I am not certain if the hr tag css I changed in NestedPanel.vue is the correct css. 

I think it sounds reasonable to change the CSS in NestedPanel, but any reason why `!important` is used and are there possible alternatives? An explanation for this would be useful as using !important unnecessarily might have unintended consequences.

> Since bootstrap.min.css also has a css for hr tags, but I'm not allowed to change it or it will fail tests.

Do you mean you edit the min.css file directly? what changes did you make and what tests are failing?

EssWhyy commented:
Hello @tlylt apologies for the late reply, here is the screenshot I meant to send. 

![possible panel css fix](https://github.com/MarkBind/markbind/assets/39799639/3cbc23ba-bc2b-423a-a2f6-6f83d557a972)

The use of !important tag was to override the bootstrap css for the hr tags. But I decided to remove them since the panel css should override it by default.

I initially tried to edit some hr css in markbind\packages\core-web\asset\css\bootstrap.min.css, but this leads to the markbind-cli:test failing. This probably is also not the best way to approach it since it affects all other hr tags.

There does not seem to be any panel in the user guide docs with a hr element, unlike the example in the issue which appears to be taken from the CS2103T website. Is there a specific class for the panels in CS2103T with the bottom hr lines?

tlylt commented:
> I initially tried to edit some hr css in markbind\packages\core-web\asset\css\bootstrap.min.css, but this leads to the markbind-cli:test failing. This probably is also not the best way to approach it since it affects all other hr tags.

yes I don't think you will ever need to edit the min.css file, unless upgrading bootstrap itself.

> There does not seem to be any panel in the user guide docs with a hr element, unlike the example in the issue which appears to be taken from the CS2103T website. Is there a specific class for the panels in CS2103T with the bottom hr lines?

Is this panel what you refer to as a panel with a hr element?
![image](https://github.com/MarkBind/markbind/assets/41845017/e392fd9c-34b3-4ab0-9968-6d1a35880894)

Besides the above, in the panel section, the distance between the button and the closing bottom line has been impacted. Is this intended by you? This UI change seems to not look as good when it's uniform across all panels.
![image](https://github.com/MarkBind/markbind/assets/41845017/616e81e5-0aff-42ff-b4ad-7f3c2d011f64)

Compare the before https://markbind-master.netlify.app/userguide/components/presentation#panels and after https://deploy-preview-2319--markbind-master.netlify.app/userguide/components/presentation#panels when doing your testing.

damithc commented:
@EssWhyy minor: can you update the PR title to be a more descriptive one?

EssWhyy commented:
For the latest commit I reverted my previous changes and added a new CSS selector within NestedPanel.vue to adjust the margins of the seamless panel hr directly. The effect is shown below.

![gap change](https://github.com/MarkBind/markbind/assets/39799639/bd7d45ce-d89b-45b4-ae8f-b448feb2add9)

tlylt commented on packages/vue-components/src/panels/NestedPanel.vue:
```diff
@@ -358,7 +363,7 @@ export default {
 
         .card-collapse > hr {
-            margin-top: 1.5rem;
+            margin-top: 1rem;
```
```suggestion
            margin-top: 0.5rem;
```
This seems to achieve the same effect for larger screen. With 1rem:
![image](https://github.com/MarkBind/markbind/assets/41845017/5758fd07-8439-48c6-81e6-3b5cf54914b3)
But with 0.5rem:
![image](https://github.com/MarkBind/markbind/assets/41845017/c3353243-358c-42e1-b89c-92ee2ccdfd3a)

tlylt commented on packages/vue-components/src/panels/NestedPanel.vue:
```diff
@@ -275,6 +275,11 @@ export default {
 
+    .expandable-card > .card-collapse > hr {
+        margin-top: 0.25rem;
+        margin-bottom: 0.75rem;
```
```suggestion
```
Any reason why `margin-bottom` is adjusted? Doesn't seem to play any role in achieving the outcome

tlylt reviewed (changes requested):
Thanks @EssWhyy for following up on this. Review comments given.

Reference links:
- the page that contains the panel mentioned in the issue: https://nus-cs2103-ay2223s2.github.io/website/admin/tp-expectations.html (Expectation Do an equal share of the team-tasks.)
- some seamless panel in our UG: https://markbind-master.netlify.app/index.html (More examples of generating static content)

tlylt commented:
@all-contributors please add @EssWhyy for code
